### ISSUE TITLE
power failure?

### ISSUE THREAD

**Issue by FelixPQ** [jedisct1/dnscrypt-proxy#685]
I'm running Pi-hole + VPN + DNSCRYPT on a raspberry Pi 3 and all was fine but since we had a power failure here my web browser is crawling almost a stand still. I ran a sudo systemctl status ... and it reports the cert is not been activated yet and the last line says no usable certificates found. I disconnected VPN and results are a little better but still very slow. Any suggestions on what I should do?
Thanks

**Comment by licaon-kter** [comment]
Try a different server!?

**Comment by FelixPQ** [comment]
I tried a different server and here is what I get:

sudo systemctl status -l dnscrypt-proxy@\*
â— dnscrypt-proxy@d0wn-us-ns2.service - DNSCrypt client proxy
   Loaded: loaded (/lib/systemd/system/dnscrypt-proxy@.service; disabled)
   Active: inactive (dead)
     Docs: man:dnscrypt-proxy(8)

â— dnscrypt-proxy@cs-caeast.service - DNSCrypt client proxy
   Loaded: loaded (/lib/systemd/system/dnscrypt-proxy@.service; disabled)
   Active: inactive (dead)
     Docs: man:dnscrypt-proxy(8)

â— dnscrypt-proxy@bn-ca0.service - DNSCrypt client proxy
   Loaded: loaded (/lib/systemd/system/dnscrypt-proxy@.service; disabled)
   Active: inactive (dead)
     Docs: man:dnscrypt-proxy(8)

â— dnscrypt-proxy@d0wn-us-ns4.service - DNSCrypt client proxy
   Loaded: loaded (/lib/systemd/system/dnscrypt-proxy@.service; disabled)
   Active: active (running) since Wed 2017-06-21 06:29:56 UTC; 21s ago
     Docs: man:dnscrypt-proxy(8)
 Main PID: 1173 (dnscrypt-proxy)
   CGroup: /system.slice/system-dnscrypt\x2dproxy.slice/dnscrypt-proxy@d0wn-us-ns4.service
           â””â”€1173 /usr/local/sbin/dnscrypt-proxy --resolver-name=d0wn-us-ns4 --user=dnscrypt

Jun 21 06:30:01 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:01 2017 [INFO] This certificate has not been activated yet
Jun 21 06:30:01 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:01 2017 [ERROR] No useable certificates found
Jun 21 06:30:07 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:07 2017 [INFO] Refetching server certificates
Jun 21 06:30:07 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:07 2017 [INFO] Server certificate with serial #1498034281 received
Jun 21 06:30:07 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:07 2017 [INFO] This certificate has not been activated yet
Jun 21 06:30:07 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:07 2017 [ERROR] No useable certificates found
Jun 21 06:30:16 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:16 2017 [INFO] Refetching server certificates
Jun 21 06:30:16 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:16 2017 [INFO] Server certificate with serial #1498034281 received
Jun 21 06:30:16 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:16 2017 [INFO] This certificate has not been activated yet
Jun 21 06:30:16 raspberrypi dnscrypt-proxy[1173]: Wed Jun 21 06:30:16 2017 [ERROR] No useable certificates found

I suppose I should removed the unused resolver? All these errors, it doesn't seem kosher. Is there something I can do?

Don't ask me why but it's much better using this one on my windows PC. I reconnected my VPN client on a couple device, oups! It's fine on the PC but it's a stand still on my Android tablet with VPN connected??? It seems fine when VPN is disconnected (it was ok before the power failure if helpful) Any suggestion?

Just in case, I see there is a Make uninstall target and it is also mentioned it's not sure it's reliable. In the case I need it, any experience trying to remove dnscrypt on a Pi using this?

**Comment by licaon-kter** [comment]
Do try _another_ server.
Those systemd (disabled) units mean nothing, Enable/start one of them.

**Comment by jedisct1** [comment]
Indeed, maybe your list of servers is out of date, and the server you chose now has a different IP, different key, or doesn't exist any more.

Don't even bother with systemd in order to test a server, just run something like

```
sudo dnscrypt-proxy -R dnscrypt.org-fr -a 0.0.0.0:41
```

And see if certificates are properly retrieved and if you an send queries

**Comment by FelixPQ** [comment]
I've given up, the list I used is from the dnscrypt website directly. I'm in Canada and it seems the servers offered don't work. Thanks

**Comment by licaon-kter** [comment]
So you've tried those [>100 servers](https://dnscrypt.org/dnscrypt-resolvers.html) and none work?

There are no "servers offered.... in Canada", they're globally accessible.

Looks to me that your setup (systemd services) has some issues and you don't know how to actually use it, as @jedisct1 said, run a separate instance directly via the command line and tell us that output.

**Comment by FelixPQ** [comment]
@licaon-kter No I haven't tried all the servers and yes I'm a newbie and it's very likely I don't know what I'm doing, it's why I simply quit when thing gets to complicated for me. 

But I can still read. On this page https://dnscrypt.org/dnscrypt-resolvers.html and this one, https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-resolvers.csv there are 1 or 2 servers mentioned (Babylon...) as been in Canada. I actually tried the 2 Canadian revolvers mentioned in the first link and finally succeeded by using one of the US ones but after the power failure we had here, I couldn't reach any site so I did as suggested and tried a second one from the US after bypassing the primary DNS server at the router so I could reach the internet again and enabling it afterwards. 

Next, I wanted to know what happened and I saw this  "No usable certificates found" and thought this wasn't good and we're here now.
 
Sorry I wasted your time on this and I apologized to you all and thank you for responding to my questions.
I just wanted to say it's to complicated for me and I don't think it's a good idea to waste more of your time on this. I removed DNSCrypt from my system and it's the end of it for me. Again, sorry I bothered you folks.

Thanks
